WCFDataService/Client/System/Data/Services/Client/Materialization/ODataEntityMaterializer.cs

```diff
@@ -974,7 +974,7 @@ private void MergeLists(MaterializerEntry entry, ClientPropertyAnnotation proper
             // has null; we can simply assign the collection. No merge required.
             if (entry.ShouldUpdateFromPayload &&
                 property.NullablePropertyType == list.GetType() &&
-                property.GetValue(entry.ResolvedObject) == null)
+                property.GetFieldOrPropertyValue(entry.ResolvedObject) == null)
```

Comment:
Hi Zoe,
I am considering simply removing the last part of the criteria (property.Get(FiledOrProperty)Value() == null) altogether.
This code path is (AFAIK) only called when the entry having the collection property is being initialized first time. We do not need to check for it being null as the assignement comes from a projection the user has always constructed the value we should use.
This would also remove the dependency of my other PR, what do you think?